caravel/assets/spec/javascripts/sqllab/reducers_spec.js

```diff
@@ -25,4 +26,73 @@ describe('sqlLabReducer', () => {
       expect(newState.tabHistory[1]).to.eq(newState.queryEditors[1].id);
     });
   });
+  describe('Alerts', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    it('should add one alert', () => {
+      newState = r.sqlLabReducer(state, actions.addAlert(alert));
+      expect(newState.alerts).to.have.lengthOf(1);
+    });
+    it('should remove one alert', () => {
+      newState = r.sqlLabReducer(newState, actions.removeAlert(newState.alerts[0]));
+      expect(newState.alerts).to.have.lengthOf(0);
+    });
+  });
+  describe('Query editors actions', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    const defaultQueryEditor = state.queryEditors[0];
+    const qe = Object.assign({}, defaultQueryEditor);
+    qe.id = '2ed2';
+    it('should add a query editor', () => {
+      newState = r.sqlLabReducer(state, actions.addQueryEditor(qe));
+      expect(newState.queryEditors).to.have.lengthOf(2);
+    });
+    it('should remove a query editor', () => {
+      newState = r.sqlLabReducer(newState, actions.removeQueryEditor(qe));
+      expect(newState.queryEditors).to.have.lengthOf(1);
+    });
+    it('should set q query editor active', () => {
+      newState = r.sqlLabReducer(state, actions.addQueryEditor(qe));
+      newState = r.sqlLabReducer(newState, actions.setActiveQueryEditor(defaultQueryEditor));
+      expect(newState.tabHistory[newState.tabHistory.length - 1]).equals(defaultQueryEditor.id);
+    });
+    it('should not fail while setting attributes', () => {
+      newState = r.sqlLabReducer(state, actions.queryEditorSetDb(qe, 1));
```

Comment:
tests that test one thing at a time are a good pattern that would be great to follow. it's easier/quicker to identify what is failing. so this could become 5 tests. we should also assert that the action didn't just not throw an error but changed the state in the correct way.
```
it('should set a query editor DB', () => {
  newState = r.sqlLabReducer(state, actions.queryEditorSetDb(qe, 1));
  // be explicit, add assertion
  // expect newState to have a qe object with the DB set to 1
});
```